packages/@tailwindcss-upgrade/src/template/codemods/arbitrary-value-to-bare-value.ts

```diff
@@ -0,0 +1,61 @@
+import type { Config } from 'tailwindcss'
+import { parseCandidate, type Candidate, type Variant } from '../../../../tailwindcss/src/candidate'
+import type { DesignSystem } from '../../../../tailwindcss/src/design-system'
+import { printCandidate } from '../candidates'
+
+export function arbitraryValueToBareValue(
+  designSystem: DesignSystem,
+  _userConfig: Config,
+  rawCandidate: string,
+): string {
+  for (let candidate of parseCandidate(rawCandidate, designSystem)) {
+    let clone = structuredClone(candidate)
+    let changed = false
+    for (let variant of variants(clone)) {
+      // Convert `data-[selected]` to `data-selected`
+      if (
+        variant.kind === 'functional' &&
+        variant.root === 'data' &&
+        variant.value?.kind === 'arbitrary' &&
+        !variant.value.value.includes('=')
+      ) {
+        changed = true
+        variant.value = {
+          kind: 'named',
+          value: variant.value.value,
+        }
+      }
+
+      // Convert `aria-[selected="true"]` to `aria-selected`
+      else if (
+        variant.kind === 'functional' &&
+        variant.root === 'aria' &&
+        variant.value?.kind === 'arbitrary' &&
+        variant.value.value.endsWith('="true"')
```

Comment:
Is it valid to pass all CSS attribute selectors here? e.g. :`!=` or `*=` etc. This would match more often than we want it in that case

Reply:
The goal here is to match what the user would've typed in the variant in v3 that is _equivalent_ to the variant in v4 so anything other than `="true"` would be wrong because `="true"` is what v4 uses.

Reply:
Ah but I see what @philipp-spiess means. If we have `aria-[foo*="true"]` then this could will match, but it should _not_ match because we can't safely migrate that. Will adjust and add a test.

Reply:
Haha yeah that's what I meant! This will then migrate to `aria-foo*` which seems unexpected